contracts/Polymorph.sol

```diff
@@ -57,24 +56,25 @@ contract Polymorph is IPolymorph, ERC721PresetMinterPauserAutoId, BMath, Reentra
         return _genes[tokenId];
     }
 
-    function _beforeTokenTransfer(address from, address to, uint256 tokenId) internal virtual override(ERC721PresetMinterPauserAutoId) {
-        if(whitelistAddresses[from] == true || whitelistAddresses[to] == true) {
-            ERC721PresetMinterPauserAutoId._beforeTokenTransfer(from, to, tokenId);
-            emit MarketplaceTransfer(tokenId, from, to);
-        } else {
-            uint256 oldGene = _genes[tokenId];
-            _genes[tokenId] = geneGenerator.random();
-            ERC721PresetMinterPauserAutoId._beforeTokenTransfer(from, to, tokenId);
-            emit TokenMorphed(tokenId, oldGene, _genes[tokenId]);
-        }
-    }
-
     function mint() public override payable nonReentrant {
         _tokenIdTracker.increment();
-        uint256 price = calcPolymorphPrice(_tokenIdTracker.current());
-        daoAddress.transfer(price);
-        _msgSender().transfer(msg.value.sub(price)); // Return excess
-        _mint(_msgSender(), _tokenIdTracker.current());
+
+        uint256 tokenId = _tokenIdTracker.current();
+        _genes[tokenId] = geneGenerator.random();
+
+        uint256 price = calcPolymorphPrice(tokenId);
+        
+        if(price > 0) {
```

Comment:
I think price will always be more than 0, so the check should be whether the excess amount is not 0.